components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -1127,6 +1128,27 @@ public void updateAPI(API api) throws APIManagementException, FaultGatewaysExcep
 
         Map<String, Map<String, String>> failedGateways = new ConcurrentHashMap<>();
         API oldApi = getAPI(api.getId());
+        Gson gson = new Gson();
+        Map<String, String> oldMonetizationProperties = gson.fromJson(oldApi.getMonetizationProperties().toString(),
+                HashMap.class);
+        if (!oldMonetizationProperties.isEmpty()) {
```

Comment:
This could result null pointer?

Reply:
fixed 276411f69f5e874554378f52f314185bb6bb89d5